### DISCUSSION TITLE
getting error while parsing list[int] from Fastapi query function

### DISCUSSION THREAD

**Discussion by mausam20** [fastapi/fastapi#11024] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
rom typing import List, Optional, Annotated, Union
from fastapi import Query, Path, Header

class MarketCompaniesSummariesQueryParams:
    def __init__(
        self,
        region: NaRegionEnum,
        level: MarketCompanyMarketsLevel,
        level_id: List[int] = Query(None,ge=0),
        comp_id: Annotated[List[int], Query(ge=0)] = None,
        sector_id: Optional[List[int]] = Query(None, ge=0),
      ):
```

### Description

getting bellow error while using list[int] from typing module 
 raise TypeError(
TypeError: The following constraints cannot be applied to typing.List[int]: 'ge'

![Screenshot from 2024-01-24 19-23-41](https://github.com/tiangolo/fastapi/assets/26969620/dd56e9e2-5dda-4f95-8a72-3a8fa20e0147)

### Operating System

Linux

### Operating System Details

VERSION="20.04.6 LTS (Focal Fossa)"
VERSION_ID="20.04"
VERSION_CODENAME=focal

### FastAPI Version

0.109.0

### Pydantic Version

2.5.3

### Python Version

3.10

**Comment by Kludex** [comment, marked as answer]
You want to use `min_length`, not `ge`.

**Comment by admo1** [comment]
Is this answer correct @mausam20?
Wasn't the goal here to make sure that all the elements in the list are positives, and not that the list has a non-zero length?

**Reply by mausam20** [reply]
@admo1 yes the requirement is to put conditions on all elements, 
for this you can use  List[Annotated[int, Query(ge=0)]]=Query(None,description="minimum value: 0")
for the new version simply giving the condition to the whole list doesn't work, need to provide it inside the list for the element.
